src/core/lightning/index.ts

```diff
@@ -604,103 +602,4 @@ export const LightningMixin = (superclass) =>
 
     // TODO manage the error case properly. right now there is a mix of string being return
     // or error being thrown. Not sure how this is handled by GraphQL
-
-    async updatePendingPayments(lock) {
-      const query = {
-        type: "payment",
-        pending: true,
-      }
-      const count = await ledger.getAccountTransactionsCount(this.user.accountPath, query)
-
-      if (count === 0) {
-        return
-      }
```

Comment:
this is missing and is needed

Reply:
Why is this needed? If `count === 0` then `ledgerService.listPendingPayments` won't return any payments and we won't do any processing which is the same result? Its a redundant check imo.

Reply:
Agreed, we'd just iterate over an empty list and return which amounts to the same behavoiur.

Reply:
you are locking the account, is less expensive do the count  query. 

did you read this comment?
```
// we only lock the account if there is some pending payment transaction, which would typically be unlikely
// we're doing the the Transaction.find after the lock to make sure there is no race condition
// note: there might be another design that doesn't requiere a lock at the uid level but only at the hash level,
// but will need to dig more into the cursor aspect of mongodb to see if there is a concurrency-safe way to do it.
```

Reply:
Yes this is a valid point - as a performance optimisation we should add the check and only lock if `count > 0`.

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/526/commits/2ef5e7d969157bbf501a983bb9fafef21f4054a2